src/Raven.Server/Documents/Replication/IncomingReplicationHandler.cs

```diff
@@ -1096,8 +1097,13 @@ private void AddReplicationPerformance(IncomingReplicationStatsAggregator stats)
                 _lastReplicationStats.TryDequeue(out stats);
         }
 
+        private readonly MultipleUseFlag _disposed = new MultipleUseFlag();
+
         public void Dispose()
         {
+            if (_disposed.Raise() == false)
+                return; // already disposed
```

Comment:
Wouldn't it be better to use DisposeOnce? It was created for this purpose.

Reply:
done